common/table.js

```diff
@@ -26,18 +26,17 @@
                 };
 
                 scope.gotoRowLink = function(index) {
-                    var tuple = scope.vm.page.tuples[index];
-                    var t_path = tuple.reference.location.compactPath;
-                    var chaiseURL = $window.location.href.replace($window.location.hash, '');
-                    var apps = ["recordset", "record", "record-two"]; // TODO add each app that uses record-table directive
-
-                    apps.forEach(function(element, index, array) {
-                        chaiseURL = chaiseURL.replace("/" + element + "/", '');
-                    });
-                    var path = chaiseURL + "/record/#" + UriUtils.fixedEncodeURIComponent(tuple.reference.location.catalog) + "/" + t_path;
-
-                    location.assign(path);
-
+                    try {
+                        var tuple = scope.vm.page.tuples[index];
+                        var appUrl = tuple.reference.contextualize.detailed.appLink;
+                        if (appUrl)
+                            location.assign(appUrl);
+                        else {
+                            throw new Error("Application Error: row linking undefined for " + tuple.reference.location.compactPath);
+                        }
+                    } catch (error) {
+                        ErrorService.errorPopup(error.message, error.code, "home page");
```

Comment:
We should use the `AlertService` to notify the user that something didn't work as expected.